tools/swift-inspect/Sources/swift-inspect/WindowsRemoteProcess.swift

```diff
@@ -233,38 +237,277 @@ internal final class WindowsRemoteProcess: RemoteProcess {
+        .withCString(encodedAs: UTF16.self) {
+          // Check that the dll file exists
+          var faAttributes: WIN32_FILE_ATTRIBUTE_DATA = .init()
+          guard GetFileAttributesExW($0, GetFileExInfoStandard, &faAttributes),
+                faAttributes.dwFileAttributes & DWORD(FILE_ATTRIBUTE_REPARSE_POINT) == 0 else {
+            print("\($0) doesn't exist")
+            return nil
+          }
+          // Allocate memory in the remote process
+          let szLength = SIZE_T(wcslen($0) * MemoryLayout<WCHAR>.size + 1)
+          guard let allocation = VirtualAllocEx(self.process, nil, szLength,
+                                                DWORD(MEM_COMMIT), DWORD(PAGE_READWRITE)) else {
+            print("VirtualAllocEx failed \(GetLastError())")
+            return nil
+          }
+          // Write the path in the allocated memory
+          if !WriteProcessMemory(self.process, allocation, $0, szLength, nil) {
+            print("WriteProcessMemory failed \(GetLastError())")
+            VirtualFreeEx(self.process, allocation, 0, DWORD(MEM_RELEASE))
+            return nil
+          }
+
+          return allocation
+      }
+    }
+  }
+
+  private func unloadDllAndPathRemote(
+      dwProcessId: DWORD, hKernel32: HMODULE!,
+      dllPathRemote: UnsafeMutableRawPointer) -> Bool {
+    // Get the dll module handle in the remote process to use it to
+    // unload it below.
+    // GetExitCodeThread returns a DWORD (32-bit) but the HMODULE
+    // returned from LoadLibraryW is a 64-bit pointer and may be truncated.
+    // So, search for it using the snapshot instead.
+    var hDllModule = HMODULE(bitPattern: 0)
+    let hModuleSnapshot: HANDLE =
+        CreateToolhelp32Snapshot(DWORD(TH32CS_SNAPMODULE), dwProcessId)
+    if hModuleSnapshot == INVALID_HANDLE_VALUE {
+      print("CreateToolhelp32Snapshot failed \(GetLastError())")
+      return false
+    }
+    defer { CloseHandle(hModuleSnapshot) }
+    var entry: MODULEENTRY32W = MODULEENTRY32W()
+    entry.dwSize = DWORD(MemoryLayout<MODULEENTRY32W>.size)
+    guard Module32FirstW(hModuleSnapshot, &entry) else {
+      print("Module32FirstW failed \(GetLastError())")
+      return false
+    }
+    repeat {
+      let module: String = withUnsafePointer(to: entry.szModule) {
+        $0.withMemoryRebound(
+          to: WCHAR.self,
+          capacity: MemoryLayout.size(ofValue: $0) / MemoryLayout<WCHAR>.size) {
+          String(decodingCString: $0, as: UTF16.self)
+        }
+      }
+      if module == "SwiftInspectClient.dll" {
+        hDllModule = entry.hModule
+      }
+    } while Module32NextW(hModuleSnapshot, &entry)
+    if hDllModule == HMODULE(bitPattern: 0) {
+      print("Failed to find the dll")
+      return false
+    }
+
+    // Unload the dll from the remote process
+    let unloadHookAddr = unsafeBitCast(GetProcAddress(hKernel32, "FreeLibrary"),
+                                       to: LPTHREAD_START_ROUTINE.self)
```

Comment:
Similar to above for `LoadLibraryW`, we can probably avoid the dynamic resolution of `FreeLibrary`.

Reply:
```
    let lpfnFreeLibrary:  (HMODULE?) -> Bool = FreeLibrary
    let unloadHookAddr = unsafeBitCast(lpfnFreeLibrary, to: LPTHREAD_START_ROUTINE.self)
```
causes a runtime cast failure

```
Swift/Builtin.swift:95: Fatal error: Can't unsafeBitCast between types of different sizes
Current stack trace:
0    (null)                             0x00007fffb7581100 swift_stdlib_reportFatalErrorInFile + 132
```

I'm not sure why this one doesn't work when the `LoadLibraryW` one worked